book/continuous_integration.md

```diff
@@ -0,0 +1,192 @@
+# Automating Code Quality Assurance
+
+## Introduction
+
+There are various tasks which can be automated to increase the quality of code and make development easier and less tedious. Automating the running of unit tests is especially important to ensuring trust in your pipeline or package, by ensuring that all unit tests pass before every merge.
+
+(continuous-integration)=
+## Continous Integration
+
+Continuous integration (CI) describes the practice of frequently committing changes to your code. [CI tools](CI-tools) support this working pattern by automating routine quality assurance tasks. This includes verifying that your code successfully builds or installs and that your [code tests](testing_code.md) run successfully.
+
+[CI is often linked](https://www.atlassian.com/continuous-delivery/principles/continuous-integration-vs-delivery-vs-deployment) to:
+
+* Continuous delivery - ensuring that your code is fit for use after each integration
+* Continuous deployment - automatically deploying working code into production
+
+(automate-tests)=
+### Automate tests to reduce risk of errors
+
+Tests should be run whenever you make changes to your project.
+This ensures that changes do not break the existing, intended functionality of your code.
+However, it is easy to forget to run your tests at regular intervals.
+You're already putting effort into `commit`ing your changes to a version control system regularly.
+
+And you're right to think "surely this could be automated too?"
+Continuous integration can be used to automate testing, amongst other quality assurance measures, and can be triggered when changes are made to your remote version control repository.
```

Comment:
This is really talking about workflows in Github/gitlab rather than ci.

Reply:
I've moved this section up, as this is really trying to spell out the motivation for using CICD tools